File: backend/app/schemas/two_factor.py
from pydantic import Field

from app.schemas.base import AppBaseModel


class TwoFactorSetupResponse(AppBaseModel):
    qr_code_base64: str
    secret: str


class VerifySetupRequest(AppBaseModel):
    totp_code: str = Field(min_length=6, max_length=6)


class BackupCodesResponse(AppBaseModel):
    backup_codes: list[str]


class DisableTwoFactorRequest(AppBaseModel):
    current_password: str


class PendingTwoFactorResponse(AppBaseModel):
    requires_2fa: bool = True
    pending_token: str


class LoginVerifyRequest(AppBaseModel):
    pending_token: str
    totp_code: str | None = None
    backup_code: str | None = None

File: frontend/src/types/twoFactor.ts
export interface TwoFactorSetupResponse {
  qr_code_base64: string;
  secret: string;
}

export interface VerifySetupPayload {
  totp_code: string;
}

export interface BackupCodesResponse {
  backup_codes: string[];
}

export interface DisableTwoFactorPayload {
  current_password: string;
}

export interface LoginVerifyPayload {
  pending_token: string;
  totp_code?: string;
  backup_code?: string;
}
